File: web/src/main/resources/static/js/services/voteService.js
app.service('voteService', function($q, factory) {
	var self = this;
	self.fetchByInitiativeAndLegislator = (initiativeId, partnerId)=>{		
		return $q(function(resolve, reject) {			
			factory.get("vote/initiative/"+initiativeId+"/partner/"+partnerId).then(function mySuccess(data) {						
				resolve(data);
			}, function myError(errResponse) {
				reject(errResponse);
			});			
		});
	};	
	
	self.getByInitiativeId = (initiativeId)=>{
		let data = {
				"initiativeId": initiativeId
		};
		return $q(function(resolve, reject) {			
			factory.get('vote/initiative', data).then(function mySuccess(data) {	
				console.log(data);
				resolve(data);
			}, function myError(errResponse) {
				reject(errResponse);
			});			
		});
	};
	
	self.post = (vote)=>{		
		return $q(function(resolve, reject) {			
			factory.post('vote', vote).then(function mySuccess(data) {						
				resolve(data);
			}, function myError(errResponse) {
				reject(errResponse);
			});			
		});
	};

});

File: usecases/src/main/java/com/solucionesdigitales/vote/service/vote/VoteService.java
package com.solucionesdigitales.vote.service.vote;

import java.util.List;

import com.solucionesdigitales.vote.entity.vote.Vote;

public interface VoteService {

	List<Vote> fetch();

	Vote post(Vote entity);

	Vote put(Vote entity);

	List<Vote> fetchByInitiativeAndLegislator(String initiativeId, String legislatorId);
	
	List<Vote> fetchByInitiative(String initiativeId);
	
	List<Vote> fetchByLegislator(String legislatorId);
}
